3495/0/triangle.rb

```diff
@@ -0,0 +1,77 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Style/GuardClause
+
+# triangle Project Code.
+
+# triangle analyzes the lengths of the sides of a triangle
+# (represented by a, b and c) and returns the type of triangle.
+#
+# It returns:
+#   :equilateral  if all sides are equal
+#   :isosceles    if exactly 2 sides are equal
+#   :scalene      if no sides are equal
+#
+# The tests for this method can be found in
+#   about_triangle_project.rb
+# and
+#   about_triangle_project_2.rb
+#
+# :reek:UtilityFunction
+class Triangle
+  attr_reader :side_a, :side_b, :side_c
+
+  def initialize(side_a, side_b, side_c)
+    @side_a = side_a
+    @side_b = side_b
+    @side_c = side_c
+  end
+
+  def type
+    if a_triangle?(self) || a_triangle_2?(self) || a_triangle_3?(self)
```

Comment:
It is not clear for me what means `a_triangle_2?` and `a_triangle_3?`. Please give clear names.

Also it strange why you write 'if triangle ?', and if it true you raise an exception 😅I think it should be something like `check_triangle_existence` which check everything and raise exception if something wrong.